libcob/intrinsic.c

```diff
@@ -4333,15 +4332,26 @@ cob_intr_char (cob_field *srcfield)
 	if (i < 1 || i > 256) {
 		*curr_field->data = 0;
 	} else {
-		*curr_field->data = (unsigned char)i - 1;
+		const unsigned char *col = COB_MODULE_PTR->collating_sequence;
+		unsigned char chr;
+		--i;
+		if (col == NULL) {
+			chr = (unsigned char)i;
+		} else {
+			for (chr = 255; chr > 0 && col[chr] != i; --chr);
+		}
+		*curr_field->data = chr;
 	}
 	return curr_field;
 }
 
 cob_field *
 cob_intr_ord (cob_field *srcfield)
 {
-	cob_alloc_set_field_uint ((cob_u32_t)(*srcfield->data + 1U));
+	const unsigned char *col = COB_MODULE_PTR->collating_sequence;
+	unsigned char chr = *srcfield->data;
+	int ord = (col == NULL ? chr : col[chr]) + 1U;
+	cob_alloc_set_field_uint ((cob_u32_t)ord);
```

Comment:
Can you add a switch here checking the field's type?

> If the class of argument-1 is national, the returned value is the ordinal position of argument-1 in the 
current national program collating sequence.

It is fine to  use `error_not_implemented()` in the second case.